### ISSUE TITLE
Tree does not add id field to the node div in the HTML

### ISSUE THREAD

**Issue by neo7337** [palantir/blueprint#4933]
#### Environment

- __Package version(s)__: v3.50.1
- __Browser and OS versions__: Chrome  and MacOS

#### Question

I am trying to build an app for which I am implementing the Tree View structure from the blueprintjs/core.
I am able to implement the Tree View structure fine, but for my use case I need to access each node by the ID.
On checking the HTML created i could see that the each div does not set each ID.
As per the TreeNodeProps we are required to add the id field in the contents json object, but looks like the id is not being used while creating the HTML of the tree.

Can you please help me figure out how can we access each node by the ID?

**Comment by adidahiya** [comment]
Fair point. I think it would make sense to render the `id` prop to the DOM, perhaps as a `data-node-id="..."` attribute. Accepting PRs for this change.
